docs/sigv4_get_sample.py

```diff
@@ -0,0 +1,137 @@
+###############################################################################
+#
+# PURPOSE: This code sample shows how to list MIE workflows using a 
+#   sigv4 signed request to the workflow execution REST API in the Media
+#   Insights Engine.
+#
+###############################################################################
+#
+# Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# This file is licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+# AWS Version 4 signing example for API Gateway REST API (GET)
+
+# See: http://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html
+# This version makes a POST request and passes request parameters
+# in the body (payload) of the request. Auth information is passed in
+# an Authorization header.
+import sys, os, base64, datetime, hashlib, hmac 
+import requests # pip install requests
+
+# ************* REQUEST VALUES *************
+method = 'GET'
+service = 'execute-api'
+# Host is the base URL for your REST API, where {restapi_id} is the API identifier, {region} is the Region of the API deployment. 
+host = '{restapi_id}.execute-api.{region}.amazonaws.com'
+region = '{region}'
+endpoint = 'https://{restapi_id}.execute-api.us-west-2.amazonaws.com/api/{api_name}/{method_name}'
+
+# Key derivation functions. See:
+# http://docs.aws.amazon.com/general/latest/gr/signature-v4-examples.html#signature-v4-examples-python
+def sign(key, msg):
+    return hmac.new(key, msg.encode('utf-8'), hashlib.sha256).digest()
+
+def getSignatureKey(key, dateStamp, regionName, serviceName):
+    kDate = sign(('AWS4' + key).encode('utf-8'), dateStamp)
+    kRegion = sign(kDate, regionName)
+    kService = sign(kRegion, serviceName)
+    kSigning = sign(kService, 'aws4_request')
+    return kSigning
+
+# Read AWS access key from env. variables or configuration file. Best practice is NOT
+# to embed credentials in code.
+access_key = os.environ.get('AWS_ACCESS_KEY_ID')
+secret_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
+if access_key is None or secret_key is None:
+    print('No access key is available.')
+    sys.exit()
+
+# Create a date for headers and the credential string
+t = datetime.datetime.utcnow()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. The recommended way to create an aware datetime object representing a specific timestamp in UTC is by using `now` or `fromtimestamp` methods and passing `tzinfo` to them.

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)

Reply:
We are not going to address this CodeGuru finding because we want to mirror the official AWS code sample for sigv4 signing: https://docs.aws.amazon.com/code-samples/latest/catalog/python-signv4-v4-signing-get-post.py.html.